apps/arena/lib/arena/game/skill.ex

```diff
@@ -217,15 +242,94 @@ defmodule Arena.Game.Skill do
 
       Process.send_after(self(), {:remove_effect, player.id, last_id}, effect.duration_ms)
 
-      effects =
-        get_in(game_state, [:players, player.id, :aditional_info, :effects])
-        |> Map.put(last_id, effect)
-
-      put_in(game_state, [:players, player.id, :aditional_info, :effects], effects)
+      put_in(
+        game_state,
+        [:players, player.id, :aditional_info, :effects, last_id],
+        Map.put(effect, :id, last_id)
+      )
       |> put_in([:last_id], last_id)
     end)
   end
 
+  def apply_effect_mechanic(players, game_state) do
+    updated_players =
+      Map.filter(players, fn {_, player} -> Player.alive?(player) end)
+      |> Enum.map(fn {_player_id, player} ->
+        player =
+          Enum.reduce(player.aditional_info.effects, player, fn {_effect_id, effect}, player ->
+            apply_effect_mechanic(player, effect, game_state)
+          end)
+
+        {player.id, player}
+      end)
+      |> Map.new()
+
+    Map.merge(players, updated_players)
+  end
+
+  def apply_effect_mechanic(player, effect, game_state) do
+    now = DateTime.utc_now() |> DateTime.to_unix(:millisecond)
+
+    mechanics_to_apply =
+      effect.effect_mechanics
+      |> Map.filter(fn {_name, mechanic} ->
+        now - Map.get(mechanic, :last_application_time, 0) >= mechanic.effect_delay_ms
+      end)
+      |> Enum.map(fn {name, mechanic} ->
+        {name, Map.put(mechanic, :last_application_time, now)}
+      end)
+      |> Map.new()
+
+    Enum.reduce(mechanics_to_apply, player, fn mechanic, player ->
+      do_effect_mechanics(game_state, player, effect, mechanic)
+    end)
+    |> update_in([:aditional_info, :effects, effect.id, :effect_mechanics], fn mechanics ->
+      Map.merge(mechanics, mechanics_to_apply)
+    end)
+  end
+
+  defp do_effect_mechanics(game_state, player, effect, {:pull, pull_params}) do
+    Map.get(game_state.pools, effect.owner_id)
+    |> case do
+      nil ->
+        player
+
+      pool ->
+        if pool.position != player.position do
+          direction = Physics.get_direction_from_positions(player.position, pool.position)
+
+          rust_player = Physics.move_entity_to_direction(player, direction, pull_params.force)
+
+          Map.put(rust_player, :aditional_info, player.aditional_info)
+          |> Map.put(:collides_with, player.collides_with)
```

Comment:
How confident are you that this is all missing when we get the entity back?

Reply:
This was something i had to do since we're calling rust to calculate the next position. When the player entity es returned it has all this missing